File: checkout-core/src/main/java/com/adyen/checkout/core/api/ThreadManager.java
/*
 * Copyright (c) 2020 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by caiof on 17/12/2020.
 */

package com.adyen.checkout.core.api;

import android.os.Handler;
import android.os.Looper;

import com.adyen.checkout.core.exception.NoConstructorException;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public final class ThreadManager {

    public static final Handler MAIN_HANDLER = getMainHandler();
    public static final ExecutorService EXECUTOR = Executors.newCachedThreadPool();

    private static Handler getMainHandler() {
        try {
            return new Handler(Looper.getMainLooper());
        } catch (RuntimeException e) {
            // avoid Looper class on testing
            return new Handler();
        }
    }

    private ThreadManager() {
        throw new NoConstructorException();
    }
}

File: checkout-core/src/main/java/com/adyen/checkout/core/api/ThreadManager.kt
/*
 * Copyright (c) 2020 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by caiof on 17/12/2020.
 */
package com.adyen.checkout.core.api

import android.os.Handler
import android.os.Looper
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

object ThreadManager {

    @Suppress("TooGenericExceptionCaught")
    @JvmField
    val MAIN_HANDLER = try {
        Handler(Looper.getMainLooper())
    } catch (e: RuntimeException) {
        // avoid Looper class on testing
        Handler()
    }
    @JvmField
    val EXECUTOR: ExecutorService = Executors.newCachedThreadPool()
}
